components/List/examples/supplemental/AutoLayoutListItemCell.m

```diff
@@ -0,0 +1,803 @@
+@property (strong, nonatomic) NSLayoutConstraint *trailingContainerCenterYConstraint;
+
+@property (strong, nonatomic) UIView *textContainer;
+@property (strong, nonatomic) NSLayoutConstraint *leadingViewTextContainerLeadingConstraint;
+@property (strong, nonatomic) NSLayoutConstraint *contentViewTextContainerLeadingConstraint;
+@property (strong, nonatomic) NSLayoutConstraint *textContainerTrailingConstraint;
+@property (strong, nonatomic) NSLayoutConstraint *textContainerTopConstraint;
+@property (strong, nonatomic) NSLayoutConstraint *textContainerBottomConstraint;
+
+@property (strong, nonatomic) UILabel *overlineLabel;
+@property (strong, nonatomic) NSLayoutConstraint *overlineLabelLeadingConstraint;
+@property (strong, nonatomic) NSLayoutConstraint *overlineLabelTrailingConstraint;
+@property (strong, nonatomic) NSLayoutConstraint *overlineLabelTopConstraint;
+@property (strong, nonatomic) NSLayoutConstraint *overlineLabelBottomConstraint;
+
+@property (strong, nonatomic) UILabel *titleLabel;
+@property (strong, nonatomic) NSLayoutConstraint *titleLabelLeadingConstraint;
+@property (strong, nonatomic) NSLayoutConstraint *titleLabelTrailingConstraint;
+@property (strong, nonatomic) NSLayoutConstraint *titleLabelTopConstraint;
+@property (strong, nonatomic) NSLayoutConstraint *titleLabelBottomConstraint;
+
+@property (strong, nonatomic) UILabel *detailLabel;
+@property (strong, nonatomic) NSLayoutConstraint *detailLabelLeadingConstraint;
+@property (strong, nonatomic) NSLayoutConstraint *detailLabelTrailingConstraint;
+@property (strong, nonatomic) NSLayoutConstraint *detailLabelTopConstraint;
+@property (strong, nonatomic) NSLayoutConstraint *detailLabelBottomConstraint;
+
+@end
+
+@implementation AutoLayoutListItemCell
+
+@synthesize mdc_adjustsFontForContentSizeCategory = _mdc_adjustsFontForContentSizeCategory;
+
+#pragma mark Object Lifecycle
+
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    [self commonInit];
+    return self;
+  }
+  return nil;
+}
+
+- (instancetype)initWithCoder:(NSCoder *)aDecoder {
+  self = [super initWithCoder:aDecoder];
+  if (self) {
+    [self commonInit];
+    return self;
+  }
+  return nil;
+}
+
+- (instancetype)initWithFrame:(CGRect)frame {
+  self = [super initWithFrame:frame];
+  if (self) {
+    [self commonInit];
+    return self;
+  }
+  return nil;
+
+}
+
+- (void)commonInit {
+  self.typographyScheme = [self defaultTypographyScheme];
+  [self createSupportingViews];
+  [self setUpLeadingViewContainerConstraints];
+  [self setUpTrailingViewContainerConstraints];
+  [self setUpTextContainerConstraints];
+  [self setUpOverlineLabelConstraints];
+  [self setUpTitleLabelConstraints];
+  [self setUpDetailLabelConstraints];
+}
+
+#pragma mark UICollectionViewCell Overrides
+
+-(void)prepareForReuse {
+  [super prepareForReuse];
+  self.overlineText = nil;
+  self.titleText = nil;
+  self.detailText = nil;
+  self.leadingView = nil;
```

Comment:
If there are performance concerns, this is where I would investigate hidden, over removing the view from the hierarchy.